less-builder.js

```diff
@@ -20,33 +36,63 @@ var cssInject = [
 ].join(';');
 
 exports.bundle = function (loads, compileOpts, outputOpts) {
-	var _this = this;
+    var loader = this;
+    var writeStubs = typeof outputOpts == 'undefined';
+    outputOpts = outputOpts || compileOpts;
+
+    var stubDefines = writeStubs ? loads.map(function(load) {
+      return (compileOpts.systemGlobal || 'System') + ".register('" + load.name + "', [], false, function() {});";
+    }).join('\n') : [];
 
-	var stubDefines = loads.map(function (load) {
-		return "System\.register('" + load.name + "', [], false, function() {});";
-	}).join('\n');
+    var rootURL = loader.rootURL || fromFileURL(loader.baseURL);
 
+    var outFile = loader.separateCSS ? outputOpts.outFile.replace(/\.js$/, '-from-less.css') : rootURL;
+
+    var importRegex = /@import\s+["']([^"']+)["'];/g;
+    var urlRegex = /url\(["']?([^"']+)["']?\)/g;
+console.log("rootURL", rootURL);
 	var lessOutput = loads.map(function (load) {
-		return load.source;
-	}).reduce(function (sourceA, sourceB) {
+        var loadAddress = fromFileURL(load.address);
+        var loadDirname = path.dirname(loadAddress);
+        var relativeLoadDirname = loadDirname.replace(rootURL, "");
+console.log("load address", loadAddress);
+		return load.source
+            .replace(importRegex, function(match, importUrl) {
+    console.log("import url", importUrl);
+                return match.replace(importUrl, loadDirname+"/"+importUrl);
+            })
+            .replace(urlRegex, function(match, url) {
+    console.log("resource url", url);
+                return match.replace(url, relativeLoadDirname+"/"+url);
+            });
+	})
+    .reduce(function (sourceA, sourceB) {
 		return sourceA + sourceB;
 	}, '');
 
-	if (_this._nodeRequire) {
-		require = _this._nodeRequire;
+	if (loader._nodeRequire) {
+		require = loader._nodeRequire;
 		try {
-			var less = _this._nodeRequire('less');
-			return new Promise(function (resolve, reject) {
-				less.render(lessOutput, {
-						compress: false
-					})
-					.then(function (data) {
-						var cssOutput = data.css;
-
-						resolve([stubDefines, cssInject, '("' + escape(cssOutput) + '");'].join('\n'));
-					}).catch(function (e) {
-						console.trace(e);
-					});
+			var less = loader._nodeRequire('less');
```

Comment:
I guess we could just use 
```js
require = System._nodeRequire;
try {
  var less=require('less');  
  ...
} catch (err) {
  ...
}
```

Because, according to line 1 we must have access to `System._nodeRequire` or we wouldn't have got this far, so the `if` block isn't needed. Also, if we already overrode the value of `require` we can use that from then on.